br/pkg/backup/client.go

```diff
@@ -481,12 +483,23 @@ func WriteBackupDDLJobs(metaWriter *metautil.MetaWriter, store kv.Storage, lastB
 	if err != nil {
 		return errors.Trace(err)
 	}
-	allJobs, err := ddl.GetAllDDLJobs(snapMeta)
+	backupSchemaVersion, err := snapMeta.GetSchemaVersion()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	schemaVersionManager := domain.GetDomain(se).DDL().SchemaVersionManager()
+	// the schema version txn is a different one from the update job txn, we may find a version v1 but the job txn
+	// is not commit yet, so we need get the lock to make sure the job txn is committed.
+	schemaVersionManager.LockSchemaVersion(0)
```

Comment:
This is very confusing at the first look... Please expose a dedicated method like `WaitSchemaVersionUnlocked()` instead of using `LockSchemaVersion(0)`.